packages/plugin-essentials/sources/commands/set/version.ts

```diff
@@ -94,10 +99,39 @@ export default class SetVersionCommand extends BaseCommand {
   }
 }
 
-export async function findVersion(configuration: Configuration, request: `stable` | `canary`) {
-  const data = await httpUtils.get(`https://repo.yarnpkg.com/tags`, {configuration, jsonResponse: true});
+export async function resolveRange(configuration: Configuration, request: string) {
+  const range = semverUtils.validRange(request)!;
+
+  const data: Tags = await httpUtils.get(`https://repo.yarnpkg.com/tags`, {configuration, jsonResponse: true});
+
+  const candidates = data.tags.filter(version => {
+    try {
+      const candidate = new semverUtils.SemVer(version);
+
+      // Ignore prereleases
+      if (candidate.prerelease.length > 0)
+        return false;
```

Comment:
Note that when I said "I'd ignore prereleases" I was thinking "no special treatment for prerelease" - basically we could just use `satisfyWithPrereleases`. It's unlikely anyone will use anything else than `^` or similar as range, and in this context they probably want the prereleases if that's all there is.